presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMapping.java

```diff
@@ -48,7 +49,8 @@ public S3SecurityMapping(
             @JsonProperty("iamRole") Optional<String> iamRole,
             @JsonProperty("allowedIamRoles") Optional<List<String>> allowedIamRoles,
             @JsonProperty("accessKey") Optional<String> accessKey,
-            @JsonProperty("secretKey") Optional<String> secretKey)
+            @JsonProperty("secretKey") Optional<String> secretKey,
+            @JsonProperty("endpoint") Optional<String> endpoint)
```

Comment:
wasn't sure, but should this be a URI instead of a string so it can get the proper validations?

Reply:
Yeah, let's make it a `URI`. The AWS client says

> the service endpoint either with or without the protocol (e.g. https://sns.us-west-1.amazonaws.com or sns.us-west-1.amazonaws.com)

But the client will then convert it to a URI, so enforcing that up front is better.

Reply:
Looked into this a bit further. The definition for endpoint in the AWS documentation is technically not a URI if no scheme is provided, evident by the [URI spec](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Definition). In the case where no scheme is provided, the URI parses attributes in undesired ways:

```scala
// example with port
scala> val u = java.net.URI.create("foo.bar.com:9000")
scala> u.getScheme
res3: String = foo.bar.com // our host got parsed as the scheme
scala> u.getSchemeSpecificPart
res28: String = 9000 // our port got parsed as part of the scheme

// example with no port
scala> val u = java.net.URI.create("foo.bar.com")
scala> u.getPath
res30: String = foo.bar.com // host got parsed as the path
```

thus, I don't think parsing this as a URI is helpful if it's possible the user can pass an endpoint with no scheme.